File: scala/src/project_euler/Problem1.scala
package project_euler

import scala.testing.Benchmark

object Problem1 {

  def main(args: Array[String]): Unit = {

    object BM extends Benchmark {
      def run = {
        val numbers = Array.range(3, 1000)
        val sum = numbers.reduceLeft { (sum: Int, num: Int) =>
          if (num % 3 == 0 || num % 5 == 0) sum + num else sum
        }
        //233168
        print(sum)
      }
    }
    println("code ran in " + BM.runBenchmark(1).head + " ms")
  }
}

File: java/project_euler/Problem1.java
package project_euler;

/**
 * Problem 1
 * 
 * @author mattt
 */
public final class Problem1 {

    public static void main(final String[] args) {
        int sum = 0;
        for (int number = 3; number < 1000; number++) {
            if (number % 3 == 0 || number % 5 == 0) {
                sum += number;
            }
        }
        // 233168
        System.out.println(sum);
    }
}
